editor/editor_help.h

```diff
@@ -251,53 +254,91 @@ class EditorHelp : public VBoxContainer {
+	inline static HashMap<StringName, HashMap<StringName, HelpData>> doc_signal_cache;
+	inline static HashMap<StringName, HashMap<StringName, HelpData>> doc_theme_item_cache;
+
+	RichTextLabel *title = nullptr;
+	HSeparator *separator = nullptr;
+	RichTextLabel *content = nullptr;
 
-	inline static HashMap<StringName, String> doc_class_cache;
-	inline static HashMap<StringName, HashMap<StringName, String>> doc_property_cache;
-	inline static HashMap<StringName, HashMap<StringName, String>> doc_method_cache;
-	inline static HashMap<StringName, HashMap<StringName, String>> doc_signal_cache;
-	inline static HashMap<StringName, HashMap<StringName, String>> doc_theme_item_cache;
+	String symbol_class_name;
+	String symbol_type;
+	String symbol_visible_type;
+	String symbol_name;
 
-	RichTextLabel *rich_text = nullptr;
+	HelpData help_data;
+
+	bool dirty = false;
+
+	float content_min_height;
+	float content_max_height;
+
+	static HelpData _get_class_help_data(const StringName &p_class_name);
+	static HelpData _get_property_help_data(const StringName &p_class_name, const StringName &p_property_name);
+	static HelpData _get_method_help_data(const StringName &p_class_name, const StringName &p_method_name);
+	static HelpData _get_signal_help_data(const StringName &p_class_name, const StringName &p_signal_name);
+	static HelpData _get_theme_item_help_data(const StringName &p_class_name, const StringName &p_theme_item_name);
+
+	void _add_type_to_title(const DocType &p_doc_type);
+	void _update_labels();
+	void _update_content_size();
 	void _go_to_help(const String &p_what);
 	void _meta_clicked(const String &p_select);
 
-	String text;
-
 protected:
-	String doc_class_name;
-	String custom_description;
-
-	static void _bind_methods();
 	void _notification(int p_what);
+	static void _bind_methods();
 
 public:
-	String get_class_description(const StringName &p_class_name) const;
-	String get_property_description(const StringName &p_class_name, const StringName &p_property_name) const;
-	String get_method_description(const StringName &p_class_name, const StringName &p_method_name) const;
-	String get_signal_description(const StringName &p_class_name, const StringName &p_signal_name) const;
-	String get_theme_item_description(const StringName &p_class_name, const StringName &p_theme_item_name) const;
+	void parse_symbol(const String &p_symbol);
+	void set_custom_text(const String &p_type, const String &p_name, const String &p_description);
+	void prepend_description(const String &p_text);
 
-	RichTextLabel *get_rich_text() { return rich_text; }
-	void set_text(const String &p_text);
+	void set_content_height_limits(float p_min, float p_max);
 
-	EditorHelpBit();
+	EditorHelpBit(const String &p_symbol = String());
 };
 
-class EditorHelpTooltip : public EditorHelpBit {
-	GDCLASS(EditorHelpTooltip, EditorHelpBit);
+// Standard tooltips do not allow you to hover over them.
+// This class is intended as a temporary workaround.
```

Comment:
I'd say it's a proper solution, not a workaround. If you want to do more than a standard tooltip, make a more specialized one.
Though it sucks that it basically bypasses the tooltip system (returning empty Controls as custom tooltips is an *interesting* solution).

Reply:
I think a better solution would be to allow hovering over tooltips (we can make this configurable). In this case, when you hover over a tooltip control, another tooltip may appear. That is, we need a tooltip stack.